core/templates/dev/head/components/SearchBarDirective.js

```diff
@@ -151,11 +164,49 @@ oppia.directive('searchBar', [function() {
             }
 
             _updateSelectionDetails('languageCodes');
+
+            if (Boolean(urlService.getUrlParams().q)) {
+              _parseURL();
+            }
             _onSearchQueryChangeExec();
 
             _searchBarFullyLoaded = true;
           }
         );
+
+        var _parseURL = function() {
+          var urlSearch = decodeURI($window.location.search);
+          var query = urlSearch.replace('?q=', '');
+
+          // Grab language code(s) from URL, these are 2 lowercase letters.
+          var _languageCodeSelections = query.match(/"\w\w"/g);
```

Comment:
Also include error cases in the karma tests, e.g. what happens if someone manually enters a url with rubbish characters, or in the wrong format. (The behavior should be graceful -- i.e. if we can't understand the URL, just display the default search page.)